source/extensions/filters/http/wasm/wasm_filter.h

```diff
@@ -28,7 +28,13 @@ class FilterConfig : Logger::Loggable<Logger::Id::wasm> {
 
   std::shared_ptr<Context> createFilter() {
     Wasm* wasm = nullptr;
+    if (!tls_slot_->currentThreadRegistered()) {
```

Comment:
When can this happen? cc @jmarantz for TLS bits.

Reply:
When using remote wasm code, `createWasm` here will return without calling this `callback`, which I think makes the `tls_slot` not yet registered when it starts serving the request:

https://github.com/envoyproxy/envoy/blob/ff13ff6d0bc031c51ef35d5b2dfc82f8388e18a1/source/extensions/filters/http/wasm/wasm_filter.cc#L15-L29


the `createWasm` logic of using remote wasm code:

https://github.com/envoyproxy/envoy/blob/ff13ff6d0bc031c51ef35d5b2dfc82f8388e18a1/source/extensions/common/wasm/wasm.cc#L409-L457